### DISCUSSION TITLE
can not show the model

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#5480] [Q&A] [answered]
now , i use a bim file load itwinjs app ui (itwinjs-core-master\test-apps\appui-test-app\standalone)
and not show any ,but i use the imodeljs(2.*) can show!
please help me ,i can send the bim file if you give me the email address!

![QQ截图20230504165346](https://user-images.githubusercontent.com/18022720/236157073-9556f921-2e2a-4b44-8de7-7ed7a2f3eefb.png)

**Comment by paprophet** [comment]
I use the code to show all model and category ,but not show any element , 
but use the same code in imodeljs 2.*  is success!
who can help me ?

code:
 .....
//m:modelarray c:categoryarray
   vp.changeModelDisplay(m, true);
   vp.changeCategoryDisplay(c, true, true);

**Comment by pmconne** [comment]
> i use a bim file load itwinjs app ui (itwinjs-core-master\test-apps\appui-test-app\standalone)

That test app was [deleted from master](https://github.com/iTwin/itwinjs-core/pull/5007) two months ago.

Please open your browser console and copy-paste all output here.

**Reply by paprophet** [reply]
i think the problem about the bim file or the app ui code between itwinjs 3.* and imodeljs 2.*
i want send the bim file to your email,and you can open the bim file use any app ui ,
use
vp.changeModelDisplay(m, true);
vp.changeCategoryDisplay(c, true, true);
code to open all model and category and find the problem.

**Reply by pmconne** [reply]
Did you open your browser console? Was there any output in it? If so, paste it here.
Have you used your browser's debugger to see how what the code does when it works (in 2.*) and when it doesn't (on master)?
Your code snippet is useless to me, since I don't know what `m` and `c` are.

**Reply by paprophet** [reply, marked as answer]
think you ,i think i solve the problem
must add the model to vp
 vp!.addViewedModels(_modelarray);
